src/govuk/components/character-count/character-count.mjs

```diff
@@ -48,13 +48,7 @@ var TRANSLATIONS_DEFAULT = {
  *
  * @class
  * @param {HTMLElement} $module - The element this component controls
- * @param {object} config - Character count config
- * @param {number} config.maxlength - The maximum number of characters. If maxwords is provided, the maxlength option will be ignored.
- * @param {number} config.maxwords - The maximum number of words. If maxwords is provided, the maxlength option will be ignored.
- * @param {number} [config.threshold=0] - The percentage value of the limit at
- * which point the count message is displayed. If this attribute is set, the
- * count message will be hidden by default.
- * @param {CharacterCountTranslations} [config.i18n = DEFAULT_TRANSLATIONS]
+ * @param {CharacterCountConfig | CharacterCountConfigWithMaxLength | CharacterCountConfigWithMaxWords} [config] - Character Count config
```

Comment:
Three mutually exclusive configs available

1. Defaults
2. Defaults with max length (but not max words)
2. Defaults with max words (but not max length)